server/interpreter.js

```diff
@@ -1106,6 +1106,33 @@ Interpreter.prototype.initArray_ = function() {
     }
   });
 
+  new this.NativeFunction({
+    id: 'Array.prototype.includes', length: 1,
+    /** @type {!Interpreter.NativeCallImpl} */
+    call: function(intrp, thread, state, thisVal, args) {
+      var searchElement = args[0];
+      var fromIndex = args[1];
+      var perms = state.scope.perms;
+      var obj = intrp.toObject(thisVal, perms);
+      var len = Interpreter.toLength(obj.get('length', perms));
+      if (len === 0) return false;
+      var n = (fromIndex === undefined ? 0 : Interpreter.toInteger(fromIndex));
+      if (n >= len) return false;
+      var k = (n >= 0) ? n : Math.max(len - Math.abs(n), 0);
+      for (; k < len; k++) {
+        if (obj.has(String(k), perms)) {
+          var v = obj.get(String(k), perms);
+          if (v === searchElement ||
+              (typeof v === 'number' && typeof searchElement === 'number' &&
+               isNaN(v) && isNaN(searchElement))) {
```

Comment:
Use `Number.isNaN` and you can omit the separate typeof tests.

Reply:
Software engineers hate this PR!  Find out why...